apps/omg_child_chain/test/omg_child_chain/integration/feed_adapter_test.exs

```diff
@@ -0,0 +1,151 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.Integration.FeedAdapterTest do
+  @moduledoc false
+
+  use ExUnitFixtures
+  use ExUnit.Case, async: false
+
+  alias FakeServer.Agents.EnvAgent
+  alias FakeServer.Env
+  alias FakeServer.HTTP.Server
+  alias OMG.ChildChain.Fees.FeedAdapter
+  alias OMG.ChildChain.Fees.JSONFeeParser
+  alias OMG.Eth
+
+  @moduletag :integration
+  @moduletag :child_chain
+
+  @server_id :fees_all_fake_server
+
+  @eth Eth.zero_address()
+  @eth_hex Eth.Encoding.to_hex(@eth)
+  @payment_tx_type OMG.WireFormatTypes.tx_type_for(:tx_payment_v1)
+
+  @initial_price 100
+  @fee %{
+    token: @eth_hex,
+    amount: @initial_price,
+    pegged_amount: 1,
+    subunit_to_unit: 1_000_000_000_000_000_000,
+    pegged_currency: "USD",
+    pegged_subunit_to_unit: 100,
+    updated_at: DateTime.from_unix!(1_546_336_800)
+  }
+
+  describe "get_fee_specs/2" do
```

Comment:
Don't get this, could you rephrase?

Reply:
your fee server or fee adapter is designed in a way that every configurable option is read _inside process state_, which means: it's considerably more prone to error in testing. instead of reading configuration from the process state, provide the configuration at boot (like name: __MODULE__).

that allows you more flexibility in testing and the process remains pure from global state.

Reply:
for example: https://github.com/omisego/elixir-omg/blob/master/apps/omg_child_chain/test/omg_child_chain/ethereum_event_aggregator_test.exs

Reply:
Yes please, I left a comment on the same topic above, switching to injected options would make these tests much simpler 🕺